src/client.rs

```diff
@@ -46,6 +53,26 @@ impl ChatGPT {
         Ok(Self { client, config })
     }
 
+    /// Constructs a new ChatGPT API client with provided API Key and Configuration
+    pub fn new_with_proxy_andconfig<S: Into<String>>(
```

Comment:
Maybe the name could be changed to something simpler, like:
```suggestion
    pub fn new_with_config_proxy<S: Into<String>>(
```